pywbem_mock/_baserepository.py

```diff
@@ -0,0 +1,409 @@
+#
+# (C) Copyright 2020 InovaDevelopment.comn
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+#
+# Author: Karl  Schopmeyer <inovadevelopment.com>
+#
+
+
+"""
+Base abscract classes for an object store for collections of
+:class:`~pywbem.CIMClass`, :class:`~pywbem.CIMInstance`, and
+:class:`~pywbem.CIMQualifierDeclaration`s and a generic API for a CIM
+repository to access and manage the collections of each of these CIM object
+types.
+
+The repository is organized by namespace such that a
+namespace must be created before it can be used to create CIM object collections
+and each namespace contains a collection of each of these CIM objects.
+"""
+
+from abc import abstractmethod, abstractproperty
+from six import add_metaclass, PY2
+from custom_inherit import DocInheritMeta
+
+
+def compatibleabstractproperty(func):
+    """
+    Python 2 and python 3 differ in decorator for abstract property.
+    in python 3 (gt 3.3) it is:
+        @property
+        @abstractproperty
+    in python 2
+        @abstractproperty
+    """
+
+    if PY2:  # pylint: disable=no-else-return
+        return abstractproperty(func)
+    else:
+        return property(abstractmethod(func))
+
+
+@add_metaclass(DocInheritMeta(style="google", abstract_base_class=True))
+class BaseObjectStore(object):
+    """
+    An abstract class that defines the APIs for the methods of an object store
+    for CIM objects including CIMClass, CIMInstance, and CIMQualifierDeclaration
+    objectsthat constitute a WBEM server repository.  This
+    class provides the abstract methods for creating, accessing, and deleting,
+    CIM objects of a single CIM object type in the repository.
+
+    An object store stores only a single CIM object type.
+    """
+
+    def __init__(self, cim_object_type):
+        """
+        Parameters:
+
+            cim_object_type(:term:`CIM object`):
+              The Pywbem cim object type as defined in cim_types.py for the
+              objects in the data store. Used to verify values on create.
+        """
+        self._cim_object_type = cim_object_type
+
+    @abstractmethod
+    def exists(self, name):
+        """
+        Test if the CIM object defined by name exists in the object store.
```

Comment:
A missed case of "defined by name" that should be changed to "identified by name".

Reply:
DONE in new commit.